girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -103,6 +108,15 @@ def __init__(self, assetstore):
             self.unavailable = True
             logger.error('Could not write to assetstore root: %s', self.assetstore['root'])
 
+        unavailable = False
+        try:
+            unavailable = getattr(self, 'unavailable')
+        except AttributeError:
+            pass
+        finally:
+            if not unavailable:
+                FilesystemAssetstoreAdapter.initialized = True
```

Comment:
@finetjul  Rather than adding this logic, I think we should just move the `mkdir` call to only be where it's actually needed. Additionally, we can make `tempDir` a `property` of the class, and thereby remove this `__init__` method altogether.